src/software/new_geom/line.cpp

```diff
@@ -0,0 +1,55 @@
+#include "software/new_geom/line.h"
+
+Line::Line() {}
+
+Line::Line(const Point &first, const Point &second) : first(first), second(second)
+{
+    calculateLine(first, second);
+}
+
+Point Line::getFirst() const
+{
+    return first;
+}
+
+Point Line::getSecond() const
+{
+    return second;
+}
+
+double Line::getSlope() const
+{
+    return (second.y() - first.y()) / (second.x() - first.x());
+}
+
+void Line::setFirst(const Point &first)
```

Comment:
would suggest changing to `setFirstPoint` just to be super clear, same goes for `setSecond` below.

Reply:
No longer relevant